adapters/repos/db/sortby.go

```diff
@@ -0,0 +1,272 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2022 SeMI Technologies B.V. All rights reserved.
+//
+//  CONTACT: hello@semi.technology
+//
+
+package db
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/semi-technologies/weaviate/entities/models"
+	"github.com/semi-technologies/weaviate/entities/schema"
+	"github.com/semi-technologies/weaviate/entities/storobj"
+)
+
+type sortBy struct {
+	objects   []*storobj.Object
+	distances []float32
+	property  string
+	dataType  []string
+	order     string
+}
+
+func (s sortBy) Len() int {
+	return len(s.objects)
+}
+
+func (s sortBy) Less(i, j int) bool {
+	return s.LessBy(i, j)
+}
+
+func (s sortBy) Swap(i, j int) {
+	s.objects[i], s.objects[j] = s.objects[j], s.objects[i]
+	if s.distances != nil {
+		s.distances[i], s.distances[j] = s.distances[j], s.distances[i]
+	}
+}
+
+func (s sortBy) LessBy(i, j int) bool {
```

Comment:
Is this the same idea as `LessThan`? It appears to be so, but not 100% sure based on this name. Maybe I'm missing some context

Reply:
I just wanted to have a clean look of the `sort Interface`
```
func (s sortBy) Len() int {
	return len(s.objects)
}

func (s sortBy) Less(i, j int) bool {
	return s.LessBy(i, j)
}

func (s sortBy) Swap(i, j int) {
	s.objects[i], s.objects[j] = s.objects[j], s.objects[i]
	if s.distances != nil {
		s.distances[i], s.distances[j] = s.distances[j], s.distances[i]
	}
}
```
so that it would be visible with one glance how the `sort interface` is implemented, and the second thing is that I wanted to make this `LessBy` logic separated to a method that I could make then further refactor